mmtk/src/julia_scanning.rs

```diff
@@ -43,7 +43,28 @@ pub unsafe fn scan_julia_object<EV: EdgeVisitor<JuliaVMEdge>>(obj: Address, clos
     // get Julia object type
     let vt = mmtk_jl_typeof(obj);
 
-    if vt == jl_symbol_type || vt as usize == JULIA_BUFF_TAG {
+    if vt as usize == JULIA_BUFF_TAG {
+        let as_tagged_value =
+            obj.as_usize() - std::mem::size_of::<crate::julia_scanning::mmtk_jl_taggedvalue_t>();
+        let t_header = Address::from_usize(as_tagged_value).load::<Address>();
+        let tag = t_header.as_usize() & 3;
+        if tag == 2 {
```

Comment:
I can see from the Julia code, that the write barrier for `jl_binding_t` sets the bits to 2. It means if `jl_binding_t` gets modified, the write barrier will set the bits to 2, and we will trace the fields in the `jl_binding_t`. You probably explained to me earlier why we need to do this, and why we don't need to trace fields in `jl_binding_t` if it is not in the remset. But I forgot the actual reason.

Can you write it into the comments why we need this code?

Reply:
Is the bits cleared in any way? If not, that means once we sets the bits for a `jl_binding_t` object, the bits stay as 2, and we will trace its fields in every subsequent GCs.

Reply:
> I can see from the Julia code, that the write barrier for `jl_binding_t` sets the bits to 2. It means if `jl_binding_t` gets modified, the write barrier will set the bits to 2, and we will trace the fields in the `jl_binding_t`. You probably explained to me earlier why we need to do this, and why we don't need to trace fields in `jl_binding_t` if it is not in the remset. But I forgot the actual reason.
> 
> Can you write it into the comments why we need this code?

Oh I see. We don't scan buffers, as they will be scanned as a part of its parent object. But when a `jl_binding_t` buffer is inserted into remset, they have be to scanned. The gc bits tells us if the buffer is in the remset. Can you put this into comments?

Reply:
> Is the bits cleared in any way? If not, that means once we sets the bits for a `jl_binding_t` object, the bits stay as 2, and we will trace its fields in every subsequent GCs.

I've added a `FIXME` for now, but that's true. We should reset the bits after the `jl_binding_t` object has been scanned.

Reply:
Why not just clear the bits after the `process_edge` calls? Are you worrying that they will cause other issues? If they do (for example, if those bits are used by others for other purposes), it means the implementation is not sound.